// @aws-sdk/client-personalize-events/dist-cjs/index.js
const { awsEndpointFunctions, emitWarningIfUnsupportedVersion: emitWarningIfUnsupportedVersion$1, createDefaultUserAgentProvider, NODE_APP_ID_CONFIG_OPTIONS, getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin } = require("@smithy/core");
const { normalizeProvider, getSmithyContext, ServiceException, NoOpLogger, emitWarningIfUnsupportedVersion, loadConfigsForDefaultMode, getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, makeBuilder, createAggregatedClient } = require("@smithy/core/client");
const { Command: $Command } = require("@smithy/core/client");
exports.$Command = $Command;
exports.__Client = Client;
const { resolveDefaultsModeConfig, loadConfig, NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS, resolveRegionConfig } = require("@smithy/core/config");
const { BinaryDecisionDiagram, EndpointCache, decideEndpoint, customEndpointFunctions, resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { parseUrl, getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { DEFAULT_RETRY_MODE, NODE_RETRY_MODE_CONFIG_OPTIONS, NODE_MAX_ATTEMPT_CONFIG_OPTIONS, resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { TypeRegistry, getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveAwsSdkSigV4Config, AwsSdkSigV4Signer, NODE_AUTH_SCHEME_PREFERENCE_OPTIONS } = require("@aws-sdk/core/httpAuthSchemes");
const { defaultProvider } = require("@aws-sdk/credential-provider-node");
const { toUtf8, fromUtf8, toBase64, fromBase64, calculateBodyLength } = require("@smithy/core/serde");
const { streamCollector, NodeHttpHandler } = require("@smithy/node-http-handler");
const { AwsRestJsonProtocol } = require("@aws-sdk/core/protocols");
const { Sha256 } = require("@smithy/core/checksum");

const defaultPersonalizeEventsHttpAuthSchemeParametersProvider = async (config, context, input) => {
    return {
        operation: getSmithyContext(context).operation,
        region: await normalizeProvider(config.region)() || (() => {
            throw new Error("expected `region` to be configured for `aws.auth#sigv4`");
        })(),
    };
};
function createAwsAuthSigv4HttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4",
        signingProperties: {
            name: "personalize",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
const defaultPersonalizeEventsHttpAuthSchemeProvider = (authParameters) => {
    const options = [];
    switch (authParameters.operation) {
        default: {
            options.push(createAwsAuthSigv4HttpAuthOption(authParameters));
        }
    }
    return options;
};
const resolveHttpAuthSchemeConfig = (config) => {
    const config_0 = resolveAwsSdkSigV4Config(config);
    return Object.assign(config_0, {
        authSchemePreference: normalizeProvider(config.authSchemePreference ?? []),
    });
};

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "personalize",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

var version = "3.1089.0";
var packageInfo = {
	version: version};

const k = "ref";
const a = -1, b = true, c = "isSet", d = "PartitionResult", e = "booleanEquals", f = "getAttr", g = { [k]: "Endpoint" }, h = { [k]: d }, i = {}, j = [{ [k]: "Region" }];
const _data = {
    conditions: [
        [c, [g]],
        [c, j],
        ["aws.partition", j, d],
        [e, [{ [k]: "UseFIPS" }, b]],
        [e, [{ [k]: "UseDualStack" }, b]],
        [e, [{ fn: f, argv: [h, "supportsDualStack"] }, b]],
        [e, [{ fn: f, argv: [h, "supportsFIPS"] }, b]]
    ],
    results: [
        [a],
        [a, "Invalid Configuration: FIPS and custom endpoint are not supported"],
        [a, "Invalid Configuration: Dualstack and custom endpoint are not supported"],
        [g, i],
        ["https://personalize-events-fips.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "FIPS and DualStack are enabled, but this partition does not support one or both"],
        ["https://personalize-events-fips.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "FIPS is enabled but this partition does not support FIPS"],
        ["https://personalize-events.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "DualStack is enabled but this partition does not support DualStack"],
        ["https://personalize-events.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "Invalid Configuration: Missing Region"]
    ]
};
const root = 2;
const r = 100_000_000;
const nodes = new Int32Array([
    -1, 1, -1,
    0, 12, 3,
    1, 4, r + 11,
    2, 5, r + 11,
    3, 8, 6,
    4, 7, r + 10,
    5, r + 8, r + 9,
    4, 10, 9,
    6, r + 6, r + 7,
    5, 11, r + 5,
    6, r + 4, r + 5,
    3, r + 1, 13,
    4, r + 2, r + 3,
]);
const bdd = BinaryDecisionDiagram.from(nodes, root, _data.conditions, _data.results);

const cache = new EndpointCache({
    size: 50,
    params: ["Endpoint", "Region", "UseDualStack", "UseFIPS"],
});
const defaultEndpointResolver = (endpointParams, context = {}) => {
    return cache.get(endpointParams, () => decideEndpoint(bdd, {
        endpointParams: endpointParams,
        logger: context.logger,
    }));
};
customEndpointFunctions.aws = awsEndpointFunctions;

class PersonalizeEventsServiceException extends ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, PersonalizeEventsServiceException.prototype);
    }
}

class InvalidInputException extends PersonalizeEventsServiceException {
    name = "InvalidInputException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInputException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInputException.prototype);
    }
}
class ResourceInUseException extends PersonalizeEventsServiceException {
    name = "ResourceInUseException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceInUseException.prototype);
    }
}
class ResourceNotFoundException extends PersonalizeEventsServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}

const _A = "Action";
const _AI = "ActionId";
const _AIL = "ActionInteractionsList";
const _AIc = "ActionInteraction";
const _AIct = "ActionImpression";
const _AL = "ActionList";
const _E = "Event";
const _EL = "EventList";
const _I = "Item";
const _II = "ItemId";
const _IIE = "InvalidInputException";
const _IL = "ItemList";
const _Im = "Impression";
const _MA = "MetricAttribution";
const _PA = "PutActions";
const _PAI = "PutActionInteractions";
const _PAIR = "PutActionInteractionsRequest";
const _PAR = "PutActionsRequest";
const _PE = "PutEvents";
const _PER = "PutEventsRequest";
const _PI = "PutItems";
const _PIR = "PutItemsRequest";
const _PU = "PutUsers";
const _PUR = "PutUsersRequest";
const _RIUE = "ResourceInUseException";
const _RNFE = "ResourceNotFoundException";
const _SJAIP = "SynthesizedJsonActionInteractionProperties";
const _SJAP = "SynthesizedJsonActionProperties";
const _SJEPJSON = "SynthesizedJsonEventPropertiesJSON";
const _SJIP = "SynthesizedJsonItemProperties";
const _SJUP = "SynthesizedJsonUserProperties";
const _U = "User";
const _UI = "UserId";
const _UL = "UserList";
const _a = "application/json";
const _aI = "actionId";
const _aIc = "actionInteractions";
const _ac = "actions";
const _c = "client";
const _dA = "datasetArn";
const _e = "error";
const _eAS = "eventAttributionSource";
const _eI = "eventId";
const _eL = "eventList";
const _eT = "eventType";
const _eV = "eventValue";
const _h = "http";
const _hE = "httpError";
const _i = "impression";
const _iI = "itemId";
const _it = "items";
const _m = "message";
const _mA = "metricAttribution";
const _mT = "mediaType";
const _p = "properties";
const _rI = "recommendationId";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.personalizeevents";
const _sA = "sentAt";
const _sI = "sessionId";
const _se = "sensitive";
const _t = "timestamp";
const _tI = "trackingId";
const _u = "users";
const _uI = "userId";
const n0 = "com.amazonaws.personalizeevents";
const _s_registry = TypeRegistry.for(_s);
var PersonalizeEventsServiceException$ = [-3, _s, "PersonalizeEventsServiceException", 0, [], []];
_s_registry.registerError(PersonalizeEventsServiceException$, PersonalizeEventsServiceException);
const n0_registry = TypeRegistry.for(n0);
var InvalidInputException$ = [-3, n0, _IIE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidInputException$, InvalidInputException);
var ResourceInUseException$ = [-3, n0, _RIUE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(ResourceInUseException$, ResourceInUseException);
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var ActionId = [0, n0, _AI, 8, 0];
var ItemId = [0, n0, _II, 8, 0];
var SynthesizedJsonActionInteractionProperties = [0, n0, _SJAIP, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonActionProperties = [0, n0, _SJAP, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonEventPropertiesJSON = [0, n0, _SJEPJSON, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonItemProperties = [0, n0, _SJIP, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonUserProperties = [0, n0, _SJUP, { [_mT]: _a, [_se]: 1 }, 0];
var UserId = [0, n0, _UI, 8, 0];
var Action$ = [3, n0, _A,
    0,
    [_aI, _p],
    [0, [() => SynthesizedJsonActionProperties, 0]], 1
];
var ActionInteraction$ = [3, n0, _AIc,
    0,
    [_aI, _sI, _t, _eT, _uI, _eI, _rI, _i, _p],
    [[() => ActionId, 0], 0, 4, 0, [() => UserId, 0], 0, 0, [() => ActionImpression, 0], [() => SynthesizedJsonActionInteractionProperties, 0]], 4
];
var Event$ = [3, n0, _E,
    8,
    [_eT, _sA, _eI, _eV, _iI, _p, _rI, _i, _mA],
    [0, 4, 0, 1, [() => ItemId, 0], [() => SynthesizedJsonEventPropertiesJSON, 0], 0, [() => Impression, 0], () => MetricAttribution$], 2
];
var Item$ = [3, n0, _I,
    0,
    [_iI, _p],
    [0, [() => SynthesizedJsonItemProperties, 0]], 1
];
var MetricAttribution$ = [3, n0, _MA,
    0,
    [_eAS],
    [0], 1
];
var PutActionInteractionsRequest$ = [3, n0, _PAIR,
    0,
    [_tI, _aIc],
    [0, [() => ActionInteractionsList, 0]], 2
];
var PutActionsRequest$ = [3, n0, _PAR,
    0,
    [_dA, _ac],
    [0, [() => ActionList, 0]], 2
];
var PutEventsRequest$ = [3, n0, _PER,
    0,
    [_tI, _sI, _eL, _uI],
    [0, 0, [() => EventList, 0], [() => UserId, 0]], 3
];
var PutItemsRequest$ = [3, n0, _PIR,
    0,
    [_dA, _it],
    [0, [() => ItemList, 0]], 2
];
var PutUsersRequest$ = [3, n0, _PUR,
    0,
    [_dA, _u],
    [0, [() => UserList, 0]], 2
];
var User$ = [3, n0, _U,
    0,
    [_uI, _p],
    [0, [() => SynthesizedJsonUserProperties, 0]], 1
];
var __Unit = "unit";
var ActionImpression = [1, n0, _AIct,
    0, [() => ActionId,
        0]
];
var ActionInteractionsList = [1, n0, _AIL,
    0, [() => ActionInteraction$,
        0]
];
var ActionList = [1, n0, _AL,
    0, [() => Action$,
        0]
];
var EventList = [1, n0, _EL,
    0, [() => Event$,
        0]
];
var Impression = [1, n0, _Im,
    0, [() => ItemId,
        0]
];
var ItemList = [1, n0, _IL,
    0, [() => Item$,
        0]
];
var UserList = [1, n0, _UL,
    0, [() => User$,
        0]
];
var PutActionInteractions$ = [9, n0, _PAI,
    { [_h]: ["POST", "/action-interactions", 200] }, () => PutActionInteractionsRequest$, () => __Unit
];
var PutActions$ = [9, n0, _PA,
    { [_h]: ["POST", "/actions", 200] }, () => PutActionsRequest$, () => __Unit
];
var PutEvents$ = [9, n0, _PE,
    { [_h]: ["POST", "/events", 200] }, () => PutEventsRequest$, () => __Unit
];
var PutItems$ = [9, n0, _PI,
    { [_h]: ["POST", "/items", 200] }, () => PutItemsRequest$, () => __Unit
];
var PutUsers$ = [9, n0, _PU,
    { [_h]: ["POST", "/users", 200] }, () => PutUsersRequest$, () => __Unit
];

const getRuntimeConfig$1 = (config) => {
    return {
        apiVersion: "2018-03-22",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultPersonalizeEventsHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsRestJsonProtocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.personalizeevents",
            errorTypeRegistries,
            version: "2018-03-22",
            serviceTarget: "AmazonPersonalizeEvents",
        },
        serviceId: config?.serviceId ?? "Personalize Events",
        sha256: config?.sha256 ?? Sha256,
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};

const getRuntimeConfig = (config) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig$1(config);
    emitWarningIfUnsupportedVersion$1(process.version);
    const loaderConfig = {
        profile: config?.profile,
        logger: clientSharedValues.logger,
    };
    return {
        ...clientSharedValues,
        ...config,
        runtime: "node",
        defaultsMode,
        authSchemePreference: config?.authSchemePreference ?? loadConfig(NODE_AUTH_SCHEME_PREFERENCE_OPTIONS, loaderConfig),
        bodyLengthChecker: config?.bodyLengthChecker ?? calculateBodyLength,
        credentialDefaultProvider: config?.credentialDefaultProvider ?? defaultProvider,
        defaultUserAgentProvider: config?.defaultUserAgentProvider ?? createDefaultUserAgentProvider({ serviceId: clientSharedValues.serviceId, clientVersion: packageInfo.version }),
        maxAttempts: config?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS, config),
        region: config?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, { ...NODE_REGION_CONFIG_FILE_OPTIONS, ...loaderConfig }),
        requestHandler: NodeHttpHandler.create(config?.requestHandler ?? defaultConfigProvider),
        retryMode: config?.retryMode ??
            loadConfig({
                ...NODE_RETRY_MODE_CONFIG_OPTIONS,
                default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE,
            }, config),
        streamCollector: config?.streamCollector ?? streamCollector,
        useDualstackEndpoint: config?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        useFipsEndpoint: config?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        userAgentAppId: config?.userAgentAppId ?? loadConfig(NODE_APP_ID_CONFIG_OPTIONS, loaderConfig),
    };
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class PersonalizeEventsClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultPersonalizeEventsHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

const command = makeBuilder(commonParams, "AmazonPersonalizeEvents", "PersonalizeEventsClient", getEndpointPlugin);
const _ep0 = {};
const _mw0 = (Command, cs, config, o) => [];

class PutActionInteractionsCommand extends command(_ep0, _mw0, "PutActionInteractions", PutActionInteractions$) {
}

class PutActionsCommand extends command(_ep0, _mw0, "PutActions", PutActions$) {
}

class PutEventsCommand extends command(_ep0, _mw0, "PutEvents", PutEvents$) {
}

class PutItemsCommand extends command(_ep0, _mw0, "PutItems", PutItems$) {
}

class PutUsersCommand extends command(_ep0, _mw0, "PutUsers", PutUsers$) {
}

const commands = {
    PutActionInteractionsCommand,
    PutActionsCommand,
    PutEventsCommand,
    PutItemsCommand,
    PutUsersCommand,
};
class PersonalizeEvents extends PersonalizeEventsClient {
}
createAggregatedClient(commands, PersonalizeEvents);

exports.Action$ = Action$;
exports.ActionInteraction$ = ActionInteraction$;
exports.Event$ = Event$;
exports.InvalidInputException = InvalidInputException;
exports.InvalidInputException$ = InvalidInputException$;
exports.Item$ = Item$;
exports.MetricAttribution$ = MetricAttribution$;
exports.PersonalizeEvents = PersonalizeEvents;
exports.PersonalizeEventsClient = PersonalizeEventsClient;
exports.PersonalizeEventsServiceException = PersonalizeEventsServiceException;
exports.PersonalizeEventsServiceException$ = PersonalizeEventsServiceException$;
exports.PutActionInteractions$ = PutActionInteractions$;
exports.PutActionInteractionsCommand = PutActionInteractionsCommand;
exports.PutActionInteractionsRequest$ = PutActionInteractionsRequest$;
exports.PutActions$ = PutActions$;
exports.PutActionsCommand = PutActionsCommand;
exports.PutActionsRequest$ = PutActionsRequest$;
exports.PutEvents$ = PutEvents$;
exports.PutEventsCommand = PutEventsCommand;
exports.PutEventsRequest$ = PutEventsRequest$;
exports.PutItems$ = PutItems$;
exports.PutItemsCommand = PutItemsCommand;
exports.PutItemsRequest$ = PutItemsRequest$;
exports.PutUsers$ = PutUsers$;
exports.PutUsersCommand = PutUsersCommand;
exports.PutUsersRequest$ = PutUsersRequest$;
exports.ResourceInUseException = ResourceInUseException;
exports.ResourceInUseException$ = ResourceInUseException$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.User$ = User$;
exports.errorTypeRegistries = errorTypeRegistries;
