// @aws-amplify/storage/dist/esm/internals/apis/uploadData.mjs
import { Amplify } from '@aws-amplify/core';
import { readFile } from '../../client/utils/readFile.mjs';
import { toBase64 } from '../../client/utils/toBase64.mjs';
import { uploadData as uploadData$1 } from '../../providers/s3/apis/internal/uploadData/index.mjs';

// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
/**
 * @internal
 */
const uploadData = (input) => {
    const { data, path, options } = input;
    return uploadData$1({ amplify: Amplify, readFile, toBase64 }, {
        path,
        data,
        options: {
            useAccelerateEndpoint: options?.useAccelerateEndpoint,
            bucket: options?.bucket,
            onProgress: options?.onProgress,
            contentDisposition: options?.contentDisposition,
            contentEncoding: options?.contentEncoding,
            contentType: options?.contentType,
            metadata: options?.metadata,
            preventOverwrite: options?.preventOverwrite,
            expectedBucketOwner: options?.expectedBucketOwner,
            checksumAlgorithm: options?.checksumAlgorithm,
            // Advanced options
            locationCredentialsProvider: options?.locationCredentialsProvider,
            customEndpoint: options?.customEndpoint,
        },
    });
};

export { uploadData };
